Engine/Source/GameObject.cpp

```diff
@@ -495,21 +503,17 @@ void GameObject::Save(Archive& archive) const {
 	}*/
 }
 
-GameObject* findGameObjectParent(GameObject* gameObject, int UID) {
-	const std::vector<GameObject*>& gameObjects = gameObject->GetChildren();
+void findGameObjectParent(const std::vector<GameObject*>& gameObjects, GameObject* gameObject) {
 	for (int i = 0; i < gameObjects.size(); i++) {
-		if (gameObjects[i]->GetID() == UID) {
-			return gameObjects[i];
+		if (gameObjects[i]->GetID() == gameObject->GetParentID()) {
+			GameObject* go = new GameObject(gameObject->GetName().c_str(), gameObject->GetID(), gameObjects[i], gameObject->GetPositionMember(), gameObject->GetScaleMember(), gameObject->GetRotationMember());
+			go->SetParentID(gameObjects[i]->GetID());
+			go->SetComponents(gameObject->GetComponents());
 		}
 		else if (gameObjects[i]->GetChildren().size() != 0) {
-			for (int j = 0; j < gameObjects[i]->GetChildren().size(); j++) {
-				findGameObjectParent(gameObjects[i]->GetChildren()[j], UID);
-			}
-
+			findGameObjectParent(gameObjects[i]->GetChildren(), gameObject);
 		}
 	}
```

Comment:
I dont understand the purpose of this method. As I understand It, It does not modify or add anything,